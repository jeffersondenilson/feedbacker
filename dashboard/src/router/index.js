import { createRouter, createWebHistory } from 'vue-router'

// route level code-splitting
// this generates a separate chunk (about.[hash].js) for this route
// which is lazy-loaded when the route is visited.

const Home = () => import('../views/Home/index')
const Feedbacks = () => import('../views/Feedbacks/index')
const Credentials = () => import('../views/Credentials/index')

const routes = [
  {
    path: '/',
    name: 'Home',
    component: Home
  },
  {
    path: '/feedbacks',
    name: 'Feedbacks',
    component: Feedbacks,
    meta: {
      hasAuth: true
    }
  },
  {
    path: '/credentials',
    name: 'Credentials',
    component: Credentials,
    meta: {
      hasAuth: true
    }
  },
  {
    path: '/:pathMatch(.*)*',
    redirect: { name: 'Home' }
  }
]

const router = createRouter({
  // process.env.BASE_URL
  history: createWebHistory('/'),
  routes
})

export default router
