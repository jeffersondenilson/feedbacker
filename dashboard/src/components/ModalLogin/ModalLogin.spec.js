import { createRouter, createWebHistory } from 'vue-router'
import { routes } from '@/router'
import Toast, { POSITION } from 'vue-toastification'
import { shallowMount } from '@vue/test-utils'
import ModalLogin from '.'
import { useField } from 'vee-validate'
jest.mock('vee-validate', () => {
  return {
    useField: jest.fn(() => {
      return { value: '', errorMessage: '' }
    })
  }
})

const router = createRouter({
  history: createWebHistory('/'),
  routes
})

describe('<ModalLogin />', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should render modal-login correctly', async () => {
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(ModalLogin, {
      global: {
        plugins: [
          router,
          [Toast, { position: POSITION.BOTTOM_RIGHT }]
        ]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should set correct value in email input', async () => {
    const email = 'valid@email.com'
    useField.mockReturnValue({ value: email, errorMessage: '' })
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(ModalLogin, {
      global: {
        plugins: [
          router,
          [Toast, { position: POSITION.BOTTOM_RIGHT }]
        ]
      }
    })

    const emailInput = wrapper.find('[data-test=email-input]')
    const emailErrorMessage = wrapper.find('[data-test=email-error-message]')
    await emailInput.setValue(email)

    expect(emailInput.element.value).toBe(email)
    expect(emailErrorMessage.exists()).toBe(false)
  })

  it('should show error with invalid email', async () => {
    const email = 'invalid.email'
    const errorMessage = '*Este campo precisa ser um e-mail válido'
    useField.mockImplementation((field, validator) => {
      if(field === 'email') {
        return { value: email, errorMessage }
      }
      return { value: '', errorMessage: '' }
    })
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(ModalLogin, {
      global: {
        plugins: [
          router,
          [Toast, { position: POSITION.BOTTOM_RIGHT }]
        ]
      }
    })

    const emailInput = wrapper.find('[data-test=email-input]')
    const emailErrorMessage = wrapper.find('[data-test=email-error-message]')
    await emailInput.setValue(email)

    expect(emailInput.element.value).toBe(email)
    expect(emailInput.classes('border-brand-danger')).toBe(true)
    expect(emailErrorMessage.exists()).toBe(true)
    expect(emailErrorMessage.text()).toBe(errorMessage)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should set correct value in password input', async () => {
    const password = '123'
    useField.mockReturnValue({ value: password, errorMessage: '' })
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(ModalLogin, {
      global: {
        plugins: [
          router,
          [Toast, { position: POSITION.BOTTOM_RIGHT }]
        ]
      }
    })

    const passwordInput = wrapper.find('[data-test=password-input]')
    const passwordErrorMessage = wrapper.find('[data-test=password-error-message]')
    await passwordInput.setValue(password)

    expect(passwordInput.element.value).toBe(password)
    expect(passwordErrorMessage.exists()).toBe(false)
  })

  it('should show error with invalid password', async () => {
    const password = '12'
    const errorMessage = '*No mínimo 3 caracteres'
    useField.mockImplementation((field, validator) => {
      if(field === 'password') {
        return { value: password, errorMessage }
      }
      return { value: '', errorMessage: '' }
    })
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(ModalLogin, {
      global: {
        plugins: [
          router,
          [Toast, { position: POSITION.BOTTOM_RIGHT }]
        ]
      }
    })

    const passwordInput = wrapper.find('[data-test=password-input]')
    const passwordErrorMessage = wrapper.find('[data-test=password-error-message]')
    await passwordInput.setValue(password)

    expect(passwordInput.element.value).toBe(password)
    expect(passwordInput.classes('border-brand-danger')).toBe(true)
    expect(passwordErrorMessage.exists()).toBe(true)
    expect(passwordErrorMessage.text()).toBe(errorMessage)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should trigger form submition', async () => {
    const spy = jest.fn()
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(ModalLogin, {
      global: {
        plugins: [
          router,
          [Toast, { position: POSITION.BOTTOM_RIGHT }]
        ],
        mocks: {
          handleSubmit: spy
        }
      }
    })
    
    const submitButton = wrapper.find('[data-test=login-submit-button]')
    await submitButton.trigger('submit')

    expect(wrapper.emitted()).toHaveProperty('submit')
    expect(spy).toHaveBeenCalledTimes(1)
  })

  it('should show loading icon when button is clicked', async () => {
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(ModalLogin, {
      global: {
        plugins: [
          router,
          [Toast, { position: POSITION.BOTTOM_RIGHT }]
        ]
      }
    })
    
    const submitButton = wrapper.find('[data-test=login-submit-button]')
    expect(submitButton.text()).toBe('Entrar')
    await submitButton.trigger('submit')

    expect(submitButton.classes('opacity-50')).toBe(true)
    expect(submitButton.attributes()).toHaveProperty('disabled')

    const loadingIcon = wrapper.find('[data-test=submit-loading-icon]')
    expect(loadingIcon.exists()).toBe(true)
    expect(submitButton.html()).toMatchSnapshot()
  })
})
