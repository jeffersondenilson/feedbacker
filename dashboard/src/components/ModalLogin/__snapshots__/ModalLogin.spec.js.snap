// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalLogin /> should render modal-login correctly 1`] = `
<div class="flex justify-between">
  <h1 class="text-xl sm:text-4xl font-black text-gray-800"> Entre na sua conta </h1><button class="text-4xl text-gray-600 focus:outline-none" data-test="close-button"> × </button>
</div>
<div class="mt-16">
  <form data-test="login-form"><label class="block"><span class="text-lg font-medium text-gray-800">E-mail</span><input type="email" class="block w-full px-4 py-3 mt-1 text-lg bg-gray-100 border-2 border-transparent rounded" placeholder="jane.doe@gmail.com" data-test="email-input">
      <!--v-if-->
    </label><label class="block mt-5"><span class="text-lg font-medium text-gray-800">Senha</span><input type="password" class="block w-full px-4 py-3 mt-1 text-lg bg-gray-100 border-2 border-transparent rounded" placeholder="******" data-test="password-input">
      <!--v-if-->
    </label><button type="submit" class="px-8 py-3 mt-10 text-2xl font-bold text-white rounded-full bg-brand-main focus:outline-none transition-all duration-150" data-test="login-submit-button"><span>Entrar</span></button></form>
</div>
`;

exports[`<ModalLogin /> should show error with invalid email 1`] = `
<div class="flex justify-between">
  <h1 class="text-xl sm:text-4xl font-black text-gray-800"> Entre na sua conta </h1><button class="text-4xl text-gray-600 focus:outline-none" data-test="close-button"> × </button>
</div>
<div class="mt-16">
  <form data-test="login-form"><label class="block"><span class="text-lg font-medium text-gray-800">E-mail</span><input type="email" class="border-brand-danger block w-full px-4 py-3 mt-1 text-lg bg-gray-100 border-2 border-transparent rounded" placeholder="jane.doe@gmail.com" data-test="email-input"><span class="block font-medium text-brand-danger" data-test="email-error-message">*Este campo precisa ser um e-mail válido</span></label><label class="block mt-5"><span class="text-lg font-medium text-gray-800">Senha</span><input type="password" class="block w-full px-4 py-3 mt-1 text-lg bg-gray-100 border-2 border-transparent rounded" placeholder="******" data-test="password-input">
      <!--v-if-->
    </label><button type="submit" class="px-8 py-3 mt-10 text-2xl font-bold text-white rounded-full bg-brand-main focus:outline-none transition-all duration-150" data-test="login-submit-button"><span>Entrar</span></button></form>
</div>
`;

exports[`<ModalLogin /> should show error with invalid password 1`] = `
<div class="flex justify-between">
  <h1 class="text-xl sm:text-4xl font-black text-gray-800"> Entre na sua conta </h1><button class="text-4xl text-gray-600 focus:outline-none" data-test="close-button"> × </button>
</div>
<div class="mt-16">
  <form data-test="login-form"><label class="block"><span class="text-lg font-medium text-gray-800">E-mail</span><input type="email" class="block w-full px-4 py-3 mt-1 text-lg bg-gray-100 border-2 border-transparent rounded" placeholder="jane.doe@gmail.com" data-test="email-input">
      <!--v-if-->
    </label><label class="block mt-5"><span class="text-lg font-medium text-gray-800">Senha</span><input type="password" class="border-brand-danger block w-full px-4 py-3 mt-1 text-lg bg-gray-100 border-2 border-transparent rounded" placeholder="******" data-test="password-input"><span class="block font-medium text-brand-danger" data-test="password-error-message">*No mínimo 3 caracteres</span></label><button type="submit" class="px-8 py-3 mt-10 text-2xl font-bold text-white rounded-full bg-brand-main focus:outline-none transition-all duration-150" data-test="login-submit-button"><span>Entrar</span></button></form>
</div>
`;

exports[`<ModalLogin /> should show loading icon when button is clicked 1`] = `
<button type="submit" class="opacity-50 px-8 py-3 mt-10 text-2xl font-bold text-white rounded-full bg-brand-main focus:outline-none transition-all duration-150" data-test="login-submit-button" disabled="">
  <icon-stub name="loading" class="animate-spin" data-test="submit-loading-icon"></icon-stub>
</button>
`;
